### DISCUSSION TITLE
Help Locating Files for New Code After RNG Changes

### DISCUSSION THREAD

**Discussion by anatolianserdar** [openssl/openssl#26224] [Q&A]
Hello,

While I was looking for some old articles and came across the crypto/rand/md_rand.c file, I noticed that it has undergone significant changes and is now totally refactored. I am trying to understand the new structure and which files or directories I should focus on to review the updated code and behavior resulting from these changes.

If anyone has insights into the refactoring or knows which other files might interact with or depend on these changes, I would appreciate your guidance.

Thanks in advance for your help!

Best regards,
Serdar

**Comment by paulidale** [comment]
Try: `providers/implementations/rands/`, `crypto/evp/evp_rand.c` and `crypto/provider_core.c` in addition to the bits in `crypto/rand/`
Tracking the flow through is fairly difficult however.
